### PR TITLE
Add random seed for reproducibility

### PR DESCRIPTION
**Pull request by aCampello**
Description
---
Partially addresses https://github.com/wellcometrust/WellcomeML/issues/140

Adds (independent) random seeds for embedding, clustering and reducing. 

It also addresses a minor issue about `kw_dictionary` not being exposed by the class.

Checklist
---

- [x] Added link to Github issue or Trello card
- [x] Added tests

### CODE REVIEW

File: wellcomeml/ml/clustering.py [hunk 1]
```diff
@@ -48,10 +50,22 @@ class that is a sklearn.base.ClusterMixin
...
+            for reducer, if applicable
+            clustering_random_state(int): Determines random number generation
+            for clustering, if applicable
+
         """
         self.embedding = embedding
         self.vectorizer = vectorizer.Vectorizer(embedding=embedding,
                                                 **params.get('embedding', {}))
+
+        if embedding_random_state and hasattr(self.vectorizer.vectorizer,
```
Comment (lizgzil) [inline wellcomeml/ml/clustering.py:65]: oh nice, I wouldnt have noticed that they might not all have one

**Review by lizgzil — APPROVED** [review]
great!

File: tests/test_clustering.py [hunk 2]
```diff
@@ -7,7 +7,10 @@
                                                      ("umap", True),
                                                      ("umap", False)])
 def test_full_pipeline(reducer, cluster_reduced):
-    cluster = TextClustering(reducer=reducer, cluster_reduced=cluster_reduced)
+    cluster = TextClustering(reducer=reducer, cluster_reduced=cluster_reduced,
+                             embedding_random_state=42,
```
Comment (nsorros) [inline tests/test_clustering.py:11]: why not one random state for all?
